src/pages/SearchPage.vue

```diff
@@ -47,42 +47,57 @@ import TheEntries from 'src/components/TheEntries.vue'
 import TheHeader from 'src/components/TheHeader.vue'
 import { useEntryStore, usePromptStore } from 'src/stores'
 import { computed, onMounted, ref } from 'vue'
+import { useRouter } from 'vue-router'
 
 const entryStore = useEntryStore()
 const promptStore = usePromptStore()
 
 const entries = ref(entryStore.getEntries)
-const categories = ref([{ label: 'All', value: 'All' }])
 const category = ref('All')
 const prompts = ref(promptStore.getPrompts)
+const router = useRouter()
 const search = ref('')
 
-onMounted(async () => {
-  await promptStore.fetchPrompts().catch((error) => errorStore.throwError(error))
-  prompts.value = promptStore.getPrompts
-
-  await entryStore.fetchEntries().catch((error) => errorStore.throwError(error))
-  entries.value = entryStore.getEntries
+onMounted(() => {
+  promptStore.fetchPrompts().catch((error) => errorStore.throwError(error))
+  entryStore.fetchEntries().catch((error) => errorStore.throwError(error))
 })
 
 promptStore.$subscribe((_mutation, state) => {
   prompts.value = state._prompts
 
-  const categoriesArr = promptStore.getPrompts.flatMap((prompt) => prompt.categories)
-  categories.value = [...new Set(categoriesArr)].map((category) => ({ label: category, value: category }))
-  categories.value.unshift({ label: 'All', value: 'All' })
+  if (router.currentRoute.value.params.year) {
+    prompts.value = prompts.value.filter((prompt) => prompt.date.split('-')[0] === router.currentRoute.value.params.year)
+  }
+})
+
+entryStore.$subscribe((_mutation, state) => {
+  entries.value = state._entries
+
+  if (router.currentRoute.value.params.year) {
+    entries.value = entries.value.filter((entry) => entry.id.split('-')[0] === router.currentRoute.value.params.year)
+  }
+})
+
+const computedCategories = computed(() => {
```

Comment:
The `computedCategories` constant is read each time the user types something into the search bar.
I implemented this so that we don't have empty categories when the user is looking for something.
Note that as you type something, categories that don't have prompts also disappear.
It looks like magic 🪄

Reply:
What I was thinking was something like this: 

in "prompts.js"
```
getters: {
  allCategories: (logic())
}

function logic() {
  const allPromptCategories = computedPrompts.value.flatMap(({ categories }) => categories)
  const uniqueCategories = Array.from(new Set(allPromptCategories), (category) => ({ label: category, value: category }))
  const allCategory = { label: 'All', value: 'All' }
 return [allCategory, ...uniqueCategories]
}
```

in SearchPage.vue
```
const computedCategories = computed(() => {
  return promptStore.allCategories
})
```

My thinking is that these two function call can be called once during set up of Pinia: 

`const allPromptCategories = computedPrompts.value.flatMap(({ categories }) => categories)`
`const uniqueCategories = Array.from(new Set(allPromptCategories), (category) => ({ label: category, value: category }))`

Reply:
I got your point. I just didn't moved it to the store because:
1. `computedCategories` uses `computedPrompts`;
2. `computedPrompts` uses `search`.

They are linked, so moving `computedCategories` to the Store would not prevent us from having a logic to deal with the remaining categories in case of a search.